File: 0001_0100/0072/0072.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = list(f.read().strip())
    l = len(s)
    k, i = l - 2, l - 1
    while s[k] >= s[k + 1]:
        k -= 1
    while s[i] <= s[k]:
        i -= 1
    s[k], s[i] = s[i], s[k]
    q.write("".join(s[:k+1] + s[k+1:][::-1]))

File: 0001_0100/0072/0072.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var s string
	fmt.Fscan(f, &s)
	l := len(s)
	t := []byte(s)
	k, i := l-2, l-1
	for t[k] >= s[k+1] {
		k--
	}
	for t[i] <= t[k] {
		i--
	}
	t[i], t[k] = t[k], t[i]
	for i = 0; i < (l-k-1)/2; i++ {
		t[k+1+i], t[l-1-i] = t[l-1-i], t[k+1+i]
	}
	fmt.Fprint(q, string(t))
}
